lib/virtio_mmio/virtio_mmio.c

```diff
@@ -29,6 +29,26 @@ extern void virtio_device_reset(struct virtio_device *vdev);
 
 void virtio_mmio_isr(struct virtio_device *vdev);
 
+#if defined(HVL_VIRTIO)
+int virtio_mmio_hvl_wait_cfg(struct virtio_device *vdev, int offset, uint32_t data);
+
+/* weak symbols */
+void virtio_mmio_shm_pool_init(void *mem, size_t size) __attribute__((weak));
```

Comment:
This seemsto be transformed to a generic callback that is called to send a signal to the guest or to the remote processor.
Similar feature is already implemented in virtio remoteproc:
https://github.com/OpenAMP/open-amp/blob/main/lib/remoteproc/remoteproc_virtio.c#L19

The callback should be provided by the system on init and stored in the the virtio_mmio_device struct